### PR TITLE
chore: Add ash version switching and CI matrix

### PR DESCRIPTION
**Pull request by TheFirstAvenger**
Add ash version switching and CI matrix same as added to `ash_postgres`

Addresses https://github.com/ash-project/ash/issues/50

### CODE REVIEW

**Comment by TheFirstAvenger** [comment]
@zachdaniel looks like ash_csv:master with ash:master is broken due to the change in DataLayer.destroy callback.

**Comment by zachdaniel** [comment]
@TheFirstAvenger that makes sense. Will probably just have to be cool w/ builds against master failing sometimes, since I don't always update the extensions every time I update master (although this one will be easy to fix).

**Comment by zachdaniel** [comment]
Before this gets merged, we'll need to make the same changes that were made here: https://github.com/ash-project/ash/commit/2ef877ee53cd0a7a1455726ce94c0ebb75989e26#diff-b6c89dbf5746803b8bea15472e8b2920L45

Otherwise, each matrix combo will try to release to hex and tweet about it 🤣

**Comment by TheFirstAvenger** [comment]
Done. I also updated to `~> 1.x.x` instead of `~> 1.x` because it was pulling `1.10.0` when specifying `~> 1.9`

**Comment by zachdaniel** [comment]
Good call 👍

File: .github/workflows/elixir.yml [hunk 1]
```diff
@@ -43,6 +46,34 @@ jobs:
         if: startsWith(github.ref, 'refs/tags/v')
       - run: mix check
         if: "!startsWith(github.ref, 'refs/tags/v')"
+                if: "!startsWith(github.ref, 'refs/tags/v')"
```
Comment (zachdaniel) [inline .github/workflows/elixir.yml]: This line looks strangely indented

**Comment by TheFirstAvenger** [comment]
@zachdaniel Looks like this one is good to go

**Comment by zachdaniel** [comment]
Excellent!

🚀 Thank you for your contribution! 🚀

**Pull request merged**